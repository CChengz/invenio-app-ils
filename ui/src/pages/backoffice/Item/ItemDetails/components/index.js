export { ItemMetadata } from './ItemMetadata';
export { ItemPastLoans } from './ItemPastLoans';
