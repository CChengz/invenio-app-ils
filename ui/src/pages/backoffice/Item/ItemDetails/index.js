import { connect } from 'react-redux';

import { deleteItem, fetchItemDetails } from './state/actions';
import ItemDetailsComponent from './ItemDetails';

const mapDispatchToProps = dispatch => ({
  fetchItemDetails: itemPid => dispatch(fetchItemDetails(itemPid)),
  deleteItem: itemPid => dispatch(deleteItem(itemPid)),
});

export const ItemDetails = connect(
  null,
  mapDispatchToProps
)(ItemDetailsComponent);
