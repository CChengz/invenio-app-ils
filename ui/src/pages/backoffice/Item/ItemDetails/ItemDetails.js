import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { Loader, Error } from '@components';
import { ItemMetadata, ItemPastLoans } from './components';

export default class ItemDetails extends Component {
  constructor(props) {
    super(props);
    this.fetchItemDetails = this.props.fetchItemDetails;
    this.deleteItem = this.props.deleteItem;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.itemPid) {
        this.fetchItemDetails(location.state.itemPid);
      }
    });
    this.fetchItemDetails(this.props.match.params.itemPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  render() {
    const { isLoading, error } = this.props;
    return (
      <Container>
        <Loader isLoading={isLoading}>
          <Error error={error}>
            <ItemMetadata />
            <ItemPastLoans />
          </Error>
        </Loader>
      </Container>
    );
  }
}

ItemDetails.propTypes = {
  fetchItemDetails: PropTypes.func.isRequired,
  deleteItem: PropTypes.func.isRequired,
};
