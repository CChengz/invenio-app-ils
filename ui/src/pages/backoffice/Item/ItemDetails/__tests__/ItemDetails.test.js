import React from 'react';
import { shallow, mount } from 'enzyme';
import ItemDetails from '../ItemDetails';

jest.mock('../components/', () => {
  return {
    ItemMetadata: () => null,
    ItemPastLoans: () => null,
  };
});

describe('ItemDetails tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      itemPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <ItemDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchItemDetails={() => {}}
        deleteItem={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch item details on mount', () => {
    const mockedFetchItemDetails = jest.fn();
    component = mount(
      <ItemDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchItemDetails={mockedFetchItemDetails}
        deleteItem={() => {}}
      />
    );
    expect(mockedFetchItemDetails).toHaveBeenCalledWith(
      routerUrlParams.params.itemPid
    );
  });
});
