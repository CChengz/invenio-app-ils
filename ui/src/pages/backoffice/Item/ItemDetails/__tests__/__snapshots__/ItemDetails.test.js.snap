// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails tests should load the details component 1`] = `
<Container>
  <Loader>
    <Error>
      <ItemMetadata />
      <ItemPastLoans />
    </Error>
  </Loader>
</Container>
`;
